### PR TITLE
Pass in-out and out arguments by reference rather than by pointer

### PR DESCRIPTION
**Pull request by eggrobin**
See #718.

### CODE REVIEW

File: benchmarks/ephemeris.cpp [hunk 1]
```diff
@@ -260,11 +260,11 @@ void EphemerisLEOProbeBenchmark(SolarSystemFactory::Accuracy const accuracy,
...
                        Norm();
     steps = trajectory.Size();
-    state->ResumeTiming();
+    state.ResumeTiming();
   }
   std::stringstream ss;
   ss << steps;
-  state->SetLabel(ss.str() + " steps, " +
+  state.SetLabel(ss.str() + " steps, " +
                   quantities::DebugString(sun_error / AstronomicalUnit) +
```
Comment (pleroy) [inline benchmarks/ephemeris.cpp]: Bad indent.

File: benchmarks/hexadecimal.cpp [hunk 2]
```diff
@@ -91,18 +91,18 @@ char constexpr π_500_bytes[] =
 
 int const copies_of_π = 10000;
 
-void HexEncode(not_null<benchmark::State*> const state,
-              not_null<bool*> const correct,
+void HexEncode(benchmark::State& state,
+              bool& correct,
```
Comment (pleroy) [inline benchmarks/hexadecimal.cpp]: Bad indent (but then it was bad before).

File: geometry/r3_element_test.cpp [hunk 3]
```diff
@@ -110,15 +110,15 @@ TEST_F(R3ElementTest, MixedProduct) {
 TEST_F(R3ElementDeathTest, OrthogonalizeError) {
   R3Element<Speed> v1 = {1 * Knot, -2 * Knot, 5 * Knot};
   EXPECT_DEATH({
-    null_velocity_.Orthogonalize<Speed>(&v1);
+    null_velocity_.Orthogonalize<Speed>(v1);
```
Comment (pleroy) [inline geometry/r3_element_test.cpp]: Is `<Speed>` needed?

File: integrators/simple_harmonic_motion.cpp [hunk 4]
```diff
@@ -205,7 +205,8 @@ TEST_P(SimpleHarmonicMotionTest, Error) {
   parameters_.sampling_period = 1;
   integrator_->SolveTrivialKineticEnergyIncrement<Length>(
```
Comment (pleroy) [inline integrators/simple_harmonic_motion.cpp]: I am also wondering if the templates parameters are needed in this file, now.

File: integrators/symmetric_linear_multistep_integrator_body.hpp [hunk 5]
```diff
@@ -213,7 +213,7 @@ StartupSolve(Instant const& t_final,
       std::min(current_state.time.value +
                    (order_ - previous_steps.size()) * step + step / 2.0,
                t_final),
-      startup_instance.get());
+      *startup_instance.get());
```
Comment (pleroy) [inline integrators/symmetric_linear_multistep_integrator_body.hpp]: You can remove `get()`, I think.

File: integrators/simple_harmonic_motion.cpp (hunk 4 above)
Comment (eggrobin) [inline integrators/simple_harmonic_motion.cpp]: I don't know, but this file is due for speedy deletion, I'm tired of the obsolete SPRK and SRKN classes.

File: integrators/symmetric_linear_multistep_integrator_body.hpp (hunk 5 above)
Comment (eggrobin) [inline integrators/symmetric_linear_multistep_integrator_body.hpp]: Done.

File: benchmarks/ephemeris.cpp (hunk 1 above)
Comment (eggrobin) [inline benchmarks/ephemeris.cpp]: Done.

**Pull request merged**